### PR TITLE
S3 and Minio Details page

### PR DESCRIPTION
**Pull request by shaik80**
### :nut_and_bolt: Description: What code changed, and why?

To support update option for S3 and minio via data-feed details page

### :chains: Related Resources
#5686 
### :+1: Definition of Done
- Test connection should be working
- Save integration should be working
- we should be able to change the bucket name and Endpoint for Minio,region for S3
### :athletic_shoe: How to Build and Test the Change
```
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

make serve

To run cypress test
cd e2e
npm run cypress:local

Type feat to enable data-feed
Navigate to settings/data-feeds
```
### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

https://user-images.githubusercontent.com/47217471/134488943-45df0286-f245-4274-b652-edd2816b9e46.mov

### CODE REVIEW

File: components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.ts [hunk 1]
```diff
@@ -111,13 +115,26 @@ export class DataFeedDetailsComponent implements OnInit, OnDestroy {
         this.destination = destination;
         this.updateForm.controls.name.setValue(this.destination.name);
         this.updateForm.controls.url.setValue(this.destination.url);
+        if (this.destination.integration_types === 'Storage') {
```
Comment (MadhviAtroliya) [inline components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.ts:118]: Can we remove the hard-coded value, put it in a variable and use it

File: components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.ts [hunk 2]
```diff
@@ -132,6 +149,31 @@ export class DataFeedDetailsComponent implements OnInit, OnDestroy {
         }
       });
   }
+  selectChangeHandlers(region: string): void {
+    this.regionSelected = region;
+  }
+  metaDataValue(): Array<KVData> {
+    if (this.destination.integration_types === 'Storage') {
```
Comment (MadhviAtroliya) [inline components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.ts:156]: Can we remove the hard-coded value, put it in a variable, and use it

Comment (MadhviAtroliya) [inline components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.ts:156]: here also ~ hardcoded value

**Comment by MadhviAtroliya** [comment]
@shaik80 please add screenshots of what you are doing in this story.
Also please check the screenshot there is a space between endpoint and bucket for Minio. 
And region dropdown is bit bigger than expected in S3.
Screenshot 2021-09-22 at 7 01 51 PM

File: components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.html [hunk 3]
```diff
@@ -74,13 +74,35 @@
...
                 Data Feed URL is required.
               </chef-error>
             </chef-form-field>
+            <chef-form-field *ngIf="destination?.services === 'S3'">
+              <label for="region">Region <span aria-hidden="true">*</span></label>
+                  <chef-select #ty [value]="regionSelected" (change)="selectChangeHandlers(ty.value)" data-cy="select-region-type" class="select-region">
+                    <chef-option *ngFor="let region of regionList" [value]="region.code"  [attr.data-cy]=" region.code ">{{ region.name }}</chef-option>
+                  </chef-select>
+            </chef-form-field>
+            <chef-form-field *ngIf="destination?.services === 'Minio'">
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.html:91]: please remove emplty element

File: components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.ts [hunk 4]
```diff
@@ -132,6 +149,31 @@ export class DataFeedDetailsComponent implements OnInit, OnDestroy {
...
+      if (this.destination.services === 'S3') {
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.ts:157]: please use enum for storage and s3

File: components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.html [hunk 5]
```diff
@@ -74,13 +74,32 @@
                 Name is required.
               </chef-error>
             </chef-form-field>
-            <chef-form-field>
-              <label for="url">Data Feed URL <span aria-hidden="true">*</span></label>
+            <chef-form-field *ngIf="!(destination?.integration_types === integrations.IntegrationTypes.STORAGE) || destination?.services === serviceIntegrations.storage.StorageIntegrationTypes.MINIO ">
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/pages/data-feed-details/data-feed-details.component.html:77]: move these logic to a function

**Comment by vivekshankar1** [comment]
Screenshot 2021-09-23 at 2 57 58 PM
The Dropdown height looks bigger than the rest of the inputs, please fix it

**Comment by shaik80** [comment]
@vivekshankar1 I have done changes Please have a look

**Review by vivekshankar1 — APPROVED** [review]
Looks fine, thanks @shaik80

**Review by MadhviAtroliya — APPROVED** [review]

**Pull request merged**